integrated_channels/integrated_channel/management/commands/transmit_courseware_data.py

```diff
@@ -0,0 +1,133 @@
+"""
+Transmits information about an enterprise's course catalog to connected IntegratedChannels
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+from logging import getLogger
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand, CommandError
+
+from enterprise.models import EnterpriseCustomer
+from integrated_channels.sap_success_factors.models import SAPSuccessFactorsEnterpriseCustomerConfiguration
+from integrated_channels.sap_success_factors.utils import SapCourseExporter
+
+try:
+    from djcelery.celery import task as celery_task
+except ImportError:
+    def celery_task(func):
+        """Use a no-op decorator if djcelery is not available."""
+        return func
+
+
+PLUGIN_MAPPING = {
+    SAPSuccessFactorsEnterpriseCustomerConfiguration: SapCourseExporter,
+}
+
+LOGGER = getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Transmit courseware data to the IntegratedChannel(s) linked to any or all EnterpriseCustomers.
+    """
+
+    def add_arguments(self, parser):
+
+        parser.add_argument(
+            '--enterprise_customer',
+            dest='enterprise_customer',
+            default=None,
+            metavar='ENTERPRISE_CUSTOMER_UUID',
+            help='Transmit data for only this EnterpriseCustomer. '
+            'Omit this option to transmit to all EnterpriseCustomers with active integrated channels.'
+        )
+        parser.add_argument(
+            '--catalog_user',
+            dest='catalog_user',
+            default=None,
+            metavar='ENTERPRISE_CATALOG_API_USERNAME',
+            help='Use this user to access the Course Catalog API. Omit this option to use the'
+            ' username specified in the EDXAPP_ENTERPRISE_CATALOG_API_USERNAME setting.'
+        )
+
+    def handle(self, *args, **options):
+        """
+        Transmit the courseware data for the EnterpriseCustomer(s) to the active integration channels.
+        """
+        enterprise_customers = []
+        # If a valid enterprise customer uuid was provided, transmit for that customer.
+        # Otherwise, transmit for all customers.
+        enterprise_customer_uuid = options['enterprise_customer']
+        if enterprise_customer_uuid:
+            try:
+                enterprise_customer = EnterpriseCustomer.active_customers.get(uuid=enterprise_customer_uuid)
+            except EnterpriseCustomer.DoesNotExist:
+                raise CommandError('Enterprise customer {} not found, or not active'.format(enterprise_customer_uuid))
+            else:
+                enterprise_customers.append(enterprise_customer)
```

Comment:
For Python, you can do a `try-except-else` block; the try will catch errors raised inside that block, and the else will allow processing if no errors are raised. It lets us be specific about the particular lines of code we want to protect.